src/autotester/server/utils/tests/file_management_test.py

```diff
@@ -0,0 +1,424 @@
+                return_value=tmp_dir,
+            ):
+                yield tmp_dir, file.name
+
+
+@pytest.fixture
+def tmp_script_dir_with_onedir():
+    """
+    Mock the test_script_directory method and yield a temporary directory and a sub directory
+    """
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        files_dir = os.path.join(tmp_dir, "files")
+        os.mkdir(files_dir)
+        with tempfile.TemporaryDirectory(dir=files_dir) as sub_sir:
+            with patch(
+                "autotester.server.utils.redis_management.test_script_directory",
+                return_value=tmp_dir,
+            ):
+                yield tmp_dir, sub_sir
+
+
+@pytest.fixture
+def nested_tmp_script_dir():
+    """
+    Mock the test_script_directory method and yield a temporary directory and nested file structure
+    """
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        files_dir = os.path.join(tmp_dir, "files")
+        os.mkdir(files_dir)
+        with tempfile.TemporaryDirectory(dir=files_dir) as sub_dir1:
+            with tempfile.TemporaryDirectory(dir=sub_dir1) as sub_dir2:
+                with tempfile.NamedTemporaryFile(dir=sub_dir2) as file:
+                    with patch(
+                        "autotester.server.utils.redis_management.test_script_directory",
+                        return_value=tmp_dir,
+                    ):
+                        yield tmp_dir, sub_dir1, sub_dir2, file.name
+
+
+def list_of_fd(file_or_dir):
+    """
+    Gets a list of files and directories and returns two separate list for files and directories
+    """
+    dir = []
+    files = []
+    for i in file_or_dir:
+        dir.append(i[1]) if i[0] == "d" else files.append(i[1])
+    return dir, files
+
+
+def test_clean_dir_name():
+    """
+    Checks whether '/' is replaced by '_' for a given path
+    """
+    a = "markus/address"
+    b = a.replace("/", "_")
+    assert clean_dir_name(a) == b
+
+
+def test_random_tmpfile_name():
+    """
+    Checks the temporary file name is random
+    """
+    tmp_file_1 = random_tmpfile_name()
+    tmp_file_2 = random_tmpfile_name()
+    assert not tmp_file_1 == tmp_file_2
+
+
+class TestRecursiveIglob:
+    """
+    Checks that all the files and directories of a given path are listed
+    """
+
+    def test_empty_dir(self, empty_dir):
+        """
+        When the Directory is empty
+        """
+        file_or_dir = list(recursive_iglob(empty_dir))
+        dir, files = list_of_fd(file_or_dir)
+        assert not dir and not files
+
+    def test_dir_has_onefile(self, dir_has_onefile):
+        """
+        When the Directory has only one file
+        """
+        root_dir, file = dir_has_onefile
+        file_or_dir = list(recursive_iglob(root_dir))
+        dir, files = list_of_fd(file_or_dir)
+        assert not dir
+        assert len(files) == 1 and file in files
+        assert os.path.exists(files[0])
```

Comment:
You should use the `all` call that you have on lines 214/215 to keep the tests consistent. Similarly for the next test below this one.